src/Runtime/Runtime/System.Windows/TextMeasurementService.cs

```diff
@@ -191,39 +211,80 @@ public bool IsTextMeasureDivID(string id)
+#endif
+            string strFontSize = (Math.Floor(fontSize * 1000) / 1000).ToString(CultureInfo.InvariantCulture) + "px";
+            string strFontFamily = fontFamily != null ? INTERNAL_FontsHelper.LoadFont(fontFamily.Source, (UIElement)associatedTextBlock) : "-";
+            string strFontStyle = style.ToString().ToLower();
+            string strFontWeight = weight.Weight.ToInvariantString();
+            string strTextWrapping = wrapping == TextWrapping.Wrap ? "pre-wrap" : "pre";
+            string strPadding = $"{padding.Top.ToInvariantString()}px {padding.Right.ToInvariantString()}px {padding.Bottom.ToInvariantString()}px {padding.Left.ToInvariantString()}px";
+            string strWidth = "";
+            string strMaxWidth = "";
             if (double.IsNaN(maxWidth) || double.IsInfinity(maxWidth))
             {
-                textBlockDivStyle.width = "";
-                textBlockDivStyle.maxWidth = "";
+                strWidth = "";
+                strMaxWidth = "";
             }
             else
             {
-                textBlockDivStyle.width = maxWidth.ToString(CultureInfo.InvariantCulture) + "px";
-                textBlockDivStyle.maxWidth = maxWidth.ToString(CultureInfo.InvariantCulture) + "px";
+                strWidth = maxWidth.ToInvariantString() + "px";
+                strMaxWidth = maxWidth.ToInvariantString() + "px";
             }
 
-            // On Simulator, it needs time to get actualwidth and actualheight
+            if (savedTextBlockFontSize == fontSize)
+                strFontSize = "";
+            else
+                savedTextBlockFontSize = fontSize;
+
+            if (savedTextBlockFontFamily == strFontFamily)
+                strFontFamily = "";
+            else
+                savedTextBlockFontFamily = strFontFamily;
+
+            if (savedTextBlockFontStyle == style)
+                strFontStyle = "";
+            else
+                savedTextBlockFontStyle = style;
+
+            if (savedTextBlockFontWeight == weight)
+                strFontWeight = "";
+            else
+                savedTextBlockFontWeight = weight;
+
+            if (savedTextBlockTextWrapping == wrapping)
+                strTextWrapping = "";
+            else
+                savedTextBlockTextWrapping = wrapping;
+
+            if (savedTextBlockPadding == padding)
+                strPadding = "";
+            else
+                savedTextBlockPadding = padding;
+
+            string javaScriptCodeToExecute = $@"document.measureTextBlock(""{strText}"",""{strFontSize}"",""{strFontFamily}"",""{strFontStyle}"",""{strFontWeight}"",""{strTextWrapping}"",""{strPadding}"",""{strWidth}"",""{strMaxWidth}"")";
 #if OPENSILVER
-            if (CSHTML5.Interop.IsRunningInTheSimulator_WorkAround)
-#else
-            if (CSHTML5.Interop.IsRunningInTheSimulator)
+            string strTextSize = Convert.ToString(CSHTML5.Interop.ExecuteJavaScript(javaScriptCodeToExecute));
```

Comment:
Use OpenSilver.Interop instead of obsolete CSHTML5.Interop
```suggestion
            string strTextSize = Convert.ToString(OpenSilver.Interop.ExecuteJavaScript(javaScriptCodeToExecute));
```